arrows/mvg/applets/init_cameras_landmarks.cxx

```diff
@@ -0,0 +1,513 @@
+    sfm_constraint_ptr(nullptr),
+    config(nullptr),
+    num_frames(0)
+  {}
+
+  camera_map_sptr camera_map_ptr;
+  landmark_map_sptr landmark_map_ptr;
+  feature_track_set_sptr feature_track_set_ptr;
+  sfm_constraints_sptr sfm_constraint_ptr;
+  initialize_cameras_landmarks_sptr algorithm;
+  kwiver::vital::config_block_sptr config;
+  size_t num_frames;
+
+  enum commandline_mode {SUCCESS, HELP, WRITE, FAIL};
+
+  commandline_mode process_command_line(cxxopts::ParseResult& cmd_args)
+  {
+    static std::string opt_config;
+    static std::string opt_out_config;
+
+    if ( cmd_args["help"].as<bool>() )
+    {
+      return HELP;
+    }
+    if ( cmd_args.count("config") )
+    {
+      opt_config = cmd_args["config"].as<std::string>();
+    }
+    if ( cmd_args.count("output-config") > 0 )
+    {
+      opt_out_config = cmd_args["output-config"].as<std::string>();
+    }
+
+    // Set up top level configuration w/ defaults where applicable.
+    config = default_config();
+
+    // If -c/--config given, read in confg file, merge in with default just
+    // generated
+    if( ! opt_config.empty() )
+    {
+      config->merge_config(kwiver::vital::read_config_file(opt_config));
+    }
+
+    bool valid_config = check_config(config);
+
+    if( ! opt_out_config.empty() )
+    {
+      write_config_file(config, opt_out_config );
+      if(valid_config)
+      {
+        LOG_INFO(main_logger,
+                 "Configuration file contained valid parameters and may be "
+                 "used for running");
+      }
+      else
+      {
+        LOG_WARN(main_logger, "Configuration deemed not valid.");
+      }
+      config = nullptr;
+      return WRITE;
+    }
+    else if(!valid_config)
+    {
+      LOG_ERROR(main_logger, "Configuration not valid.");
+      config = nullptr;
+      return FAIL;
+    }
+
+    return SUCCESS;
+  }
+
+  void register_algorithms()
+  {
+    // register the algorithm implementations
+    std::string rel_plugin_path = kwiver::vital::get_executable_path() +
+    "/../lib/kwiver/modules";
+    kwiver::vital::plugin_manager::instance().add_search_path(rel_plugin_path);
+    kwiver::vital::plugin_manager::instance().load_all_plugins();
+  }
+
+  void initialize()
+  {
+    // Create algorithm from configuration
+    initialize_cameras_landmarks::set_nested_algo_configuration( "initializer",
+                                                                 config,
+                                                                 algorithm );
+  }
+
+  void clear_ptrs()
+  {
+    camera_map_ptr = nullptr;
+    landmark_map_ptr = nullptr;
+    feature_track_set_ptr = nullptr;
+    sfm_constraint_ptr = nullptr;
+  }
+
+  void load_tracks( )
+  {
+    kwiver::vital::path_t in_tracks_path =
+      config->get_value<kwiver::vital::path_t>("input_tracks_file");
+    feature_track_set_ptr =
+      kwiver::vital::read_feature_track_file(in_tracks_path);
+  }
+
+  void load_sfm_constraint( )
```

Comment:
As is obvious here, we only need the video for its metadata in this tool.  In the future it would be better to write out the metadata map to a file during tracking and just read that back in.  TeleSculptor doesn't even do that yet, but it should.

Note that this does not need to be address now, it more of a direction for future improvement.